### PR TITLE
new dataset type and refactoring offline part of interactive_learning_approach

### PR DESCRIPTION
**Pull request by tomsilver**
I think this is mostly done, but I may have discovered a bug in interactive_learning_approach that I want to clear up before we merge (checking with @amburger66 offline if my understanding is correct).

### CODE REVIEW

File: src/datasets/ground_atom_data.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+from typing import List, Set
+from predicators.src.envs import BaseEnv
+from predicators.src.structs import Dataset, Predicate, GroundAtom
+from predicators.src import utils
+
+
+def create_ground_atom_data(env: BaseEnv, base_dataset: Dataset,
+                            annotating_predicates: Set[Predicate],
+                            annotating_ratio: float) -> Dataset:
+    """Create offline data by annotating low-level trajectories with ground
```
Comment (ronuchit) [inline src/datasets/ground_atom_data.py:13]: Can you split this into 2 sentences? It's run-on / hard to understand.

File: src/datasets/ground_atom_data.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    atoms that hold in the respective states for predicates in
+    annotating_predicates.
+
+    The base_dataset is the initial dataset (e.g. demos) of unannotated
+    low level trajectories.
```
Comment (ronuchit) [inline src/datasets/ground_atom_data.py:18]: low-level*

File: src/datasets/__init__.py [hunk 3]
```diff
@@ -18,4 +19,14 @@ def create_dataset(env: BaseEnv, train_tasks: List[Task]) -> Dataset:
         return create_demo_replay_data(env, train_tasks)
     if CFG.offline_data_method == "demo+nonoptimalreplay":
         return create_demo_replay_data(env, train_tasks, nonoptimal_only=True)
+    if CFG.offline_data_method == "demo+ground_atoms":
+        base_dataset = create_demo_data(env, train_tasks)
+        known_predicate_names = CFG.interactive_known_predicates.split(",")
+        predicates_to_learn = {
+            p
```
Comment (ronuchit) [inline src/datasets/__init__.py:26]: lol so ugly thanks yapf

File: src/approaches/interactive_learning_approach.py [hunk 4]
```diff
@@ -98,23 +89,20 @@ def learn_from_offline_dataset(self, dataset: Dataset) -> None:
...
-                    score = score_atom(dataset_with_atoms, atom)
+                    score = score_atom(dataset, atom)
                     # Ask about this atom if it is the best seen so far
                     if score > best_score:
                         if self._ask_teacher(s, atom):
                             # Add this atom if it's a positive example
                             traj = LowLevelTrajectory([s], [])
                             dataset.trajectories.append(traj)
-                            dataset_with_atoms.append((traj, [{atom}]))
+                            dataset.annotations.append([{atom}])
```
Comment (ronuchit) [inline src/approaches/interactive_learning_approach.py:99]: Would it be useful to make an `add` method for a dataset object?

**Review by ronuchit — APPROVED** [review]
lgtm, minor

**Comment by tomsilver** [comment]
confirmed the bug; fixing in #377

**Pull request merged**